client/cherami/publisher.go

```diff
@@ -68,13 +68,19 @@ const (
 var _ Publisher = (*publisherImpl)(nil)
 
 // NewPublisher constructs a new Publisher object
+// Deprecated: NewPublisher is deprecated, please use NewPublisher2
 func NewPublisher(client *clientImpl, path string, maxInflightMessagesPerConnection int) Publisher {
+	return NewPublisher2(client, path, maxInflightMessagesPerConnection, client.options.MetricsReporter)
+}
+
+// NewPublisher2 constructs a new Publisher object
+func NewPublisher2(client *clientImpl, path string, maxInflightMessagesPerConnection int, reporter metrics.Reporter) Publisher {
```

Comment:
As discussed in person, let's think of a better name as well.. NewPublisherWithReporter or something like that?